src/vcpkg/base/messages.cpp

```diff
@@ -95,17 +96,160 @@ namespace vcpkg
+                                                                     : StringLiteral("");                              \
+    };
+#include <vcpkg/base/message-args.inc.h>
+#undef DECLARE_MSG_ARG
+
+namespace vcpkg
+{
+    namespace
+    {
+        static constexpr const size_t max_number_of_args = 5;
+
+        struct MessageData
+        {
+            StringLiteral name;
+            std::array<const StringLiteral*, max_number_of_args> arg_examples;
+            StringLiteral comment;
+            StringLiteral builtin_message;
+        };
+
+        template<class... Args>
+        constexpr std::array<const StringLiteral*, max_number_of_args> make_arg_examples_array(Args...)
+        {
+            return std::array<const StringLiteral*, max_number_of_args>{&msg::ArgExample<Args>::example...};
+        }
+
+        constexpr MessageData message_data[] = {
+#define DECLARE_MESSAGE(NAME, ARGS, COMMENT, ...) {#NAME, make_arg_examples_array ARGS, COMMENT, __VA_ARGS__},
+#include <vcpkg/base/message-data.inc.h>
+#undef DECLARE_MESSAGE
+        };
+    }
+
+    namespace msg::detail
+    {
+        static constexpr const ::size_t number_of_messages = sizeof(message_data) / sizeof(message_data[0]);
+    }
+    static std::string* loaded_localization_data = 0;
+    static const char *loaded_localization_file_begin = 0, *loaded_localization_file_end = 0;
+
+    namespace
+    {
+        enum class message_index
+        {
+#define DECLARE_MESSAGE(NAME, ARGS, COMMENT, ...) NAME,
+#include <vcpkg/base/message-data.inc.h>
+#undef DECLARE_MESSAGE
+        };
+    }
+
+    namespace msg
+    {
+        namespace
+        {
+            std::string get_localization_comment(::size_t index)
+            {
+                if (index >= detail::number_of_messages) Checks::unreachable(VCPKG_LINE_INFO);
+                std::string msg = message_data[index].comment.to_string();
+                for (auto&& ex : message_data[index].arg_examples)
+                {
+                    if (ex == nullptr || ex->empty()) continue;
+                    if (!msg.empty()) msg.push_back(' ');
+                    msg.append("An example of ");
+                    msg.append(ex->data(), ex->size());
+                }
+                return msg;
+            }
+        }
+
+        std::vector<RawMessage> get_sorted_raw_messages()
+        {
+            struct MessageSorter
+            {
+                bool operator()(const RawMessage& lhs, const RawMessage& rhs) const { return lhs.name < rhs.name; }
+            };
+
+            std::vector<RawMessage> messages(msg::detail::number_of_messages);
+            for (::size_t index = 0; index < msg::detail::number_of_messages; ++index)
+            {
+                auto& msg = messages[index];
+                msg.name = message_data[index].name.to_string();
+                msg.value = message_data[index].builtin_message.to_string();
```

Comment:
Does this mean that RawMessage::name and RawMessage::value could be `StringView` or `StringLiteral`?

Reply:
Yeah, I just was doing a minimum of changes to the other code since it's enormously non-performance-sensitive. Will fix.